client/app/competitor/page.tsx

```diff
@@ -32,17 +32,25 @@ import { testColor } from '@/lib/utils';
 import { Markdown } from '@/components/Markdown';
 import { CodeBlock, Tooltip } from '@/components/util';
 import { Button } from '@/components/ui/button';
-import { currentTabAtom } from '@/lib/competitor-state';
+import { currentTabAtom, selectedLanguageAtom } from '@/lib/competitor-state';
 import { toast } from '@/hooks/use-toast';
 
 const EditorButtons = () => {
     const [currQuestion] = useAtom(currQuestionAtom);
+    const [selectedLanguage, setSelectedLanguage] = useAtom(selectedLanguageAtom);
+
     const notImplemented = () =>
         toast({
             title: 'Not Yet Implemented',
             description: 'Check back later!',
             variant: 'destructive',
         });
+
+    const handleLanguageSelection = (value: string) => {
+        setSelectedLanguage(value);
+        console.log(value);
+    };
```

Comment:
This function is unnecessary.  Remove the console lot and inline the `setSelectedLanguage`